src/outputtype.cpp

```diff
@@ -63,21 +63,34 @@ CTxDestination GetDestinationForKey(const CPubKey& key, OutputType type)
             return witdest;
         }
     }
-    case OutputType::ETH: return WitnessV16EthHash(key);
+    case OutputType::ETH: {
+        CPubKey pubkeyCopy = key;
+        if (pubkeyCopy.IsCompressed()) {
+            pubkeyCopy.Decompress();
+        }
+        return WitnessV16EthHash(pubkeyCopy);
+    }
     default: assert(false);
     }
 }
 
 std::vector<CTxDestination> GetAllDestinationsForKey(const CPubKey& key)
 {
     PKHash keyid(key);
+    auto pubkeyCopy = key;
     if (key.IsCompressed()) {
+        pubkeyCopy.Decompress();
         CTxDestination segwit = WitnessV0KeyHash(keyid);
         CTxDestination p2sh = ScriptHash(GetScriptForDestination(segwit));
-        return std::vector<CTxDestination>{std::move(keyid), std::move(p2sh), std::move(segwit)};
+        CTxDestination eth = WitnessV16EthHash(pubkeyCopy);
+        CTxDestination uncompressedLegacy = PKHash(pubkeyCopy);
+        return std::vector<CTxDestination>{std::move(keyid), std::move(p2sh), std::move(segwit), std::move(eth), std::move(uncompressedLegacy)};
     } else {
+        pubkeyCopy.Compress();
         CTxDestination eth = WitnessV16EthHash(key);
-        return std::vector<CTxDestination>{std::move(keyid), std::move(eth)};
+        CTxDestination segwit = WitnessV0KeyHash(pubkeyCopy);
+        CTxDestination p2sh = ScriptHash(GetScriptForDestination(segwit));
+        return std::vector<CTxDestination>{std::move(keyid), std::move(p2sh), std::move(segwit), std::move(eth)};
     }
```

Comment:
Can't this be:

```
if (key.IsCompressed()) {
      pubkeyCopy.Decompress();
}
```

And just not un-equivocally use uncompressed for legacy and compressed for eth and bech32 without the branching?

Reply:
Sure. Will update and use everywhere.